cmd/edit/cluster/cmd.go

```diff
@@ -552,3 +577,102 @@ func parseRFC3339(s string) (time.Time, error) {
 func isExpectedHTTPProxyOrHTTPSProxy(httpProxy, httpsProxy *string, noProxySlice []string, cluster *cmv1.Cluster) bool {
 	return httpProxy == nil && httpsProxy == nil && len(noProxySlice) > 0 && cluster.Proxy() == nil
 }
+
+func auditLogRoleExists(cluster *cmv1.Cluster) bool {
+	return cluster.AWS().AuditLog().RoleArn() != ""
+}
+
+func auditLogEnableOrUpdatePrompt(currentAuditLogRole string) string {
+	if currentAuditLogRole == "" {
+		return "Enable audit log forwarding to AWS CloudWatch"
+	}
+	return fmt.Sprintf("Update existing audit log forwarding role '%s'", currentAuditLogRole)
+}
+
+func setAuditLogForwarding(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster, auditLogArn string) (
+	argValuePtr *string, err error) {
+	if cmd.Flags().Changed("audit-log-arn") {
+		if !aws.IsHostedCP(cluster) {
+			return nil, fmt.Errorf("Audit log forwarding to AWS CloudWatch is only supported for Hosted Control Plane clusters")
+
+		}
+		if auditLogArn != "" && !aws.AuditLogArnRE.MatchString(auditLogArn) {
+			return nil, fmt.Errorf("Expected a valid value for audit-log-arn matching %s", aws.AuditLogArnRE.String())
+		}
+		argValuePtr := new(string)
+		*argValuePtr = auditLogArn
+
+		confirmAuditLogForwarding(r, argValuePtr)
+		return argValuePtr, nil
+	}
+	return nil, nil
+}
+
+func confirmAuditLogForwarding(r *rosa.Runtime, auditLogArn *string) {
+
+	if *auditLogArn != "" {
+		r.Reporter.Warnf("You are choosing to enable audit log forwarding")
+		if !confirm.Confirm("enable audit log forwarding for cluster with the provided role arn '%s'", *auditLogArn) {
+			os.Exit(0)
+		}
+		return
+	}
+	r.Reporter.Warnf("You are choosing to disable audit log forwarding.")
+	if !confirm.Confirm("disable audit log forwarding for cluster") {
+		os.Exit(0)
+	}
+}
+
+func auditLogInteractivePrompt(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster) (
+	argValuePtr *string) {
+
+	auditLogRolePtr := new(string)
+
+	if !auditLogRoleExists(cluster) {
+		requestAuditLogForwarding, err := interactive.GetBool(interactive.Input{
+			Question: auditLogEnableOrUpdatePrompt(cluster.AWS().AuditLog().RoleArn()),
+			Default:  false,
+			Required: true,
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid value: %s", err)
+			os.Exit(1)
```

Comment:
NP
We can return an error here as well instead of exiting the program.

Reply:
Thank you for the find:)